### ISSUE TITLE
Multiprocessing Hanging

### ISSUE THREAD

**Issue by william-silversmith** [seung-lab/cloud-volume#103]
```
from cloudvolume import CloudVolume

vol = CloudVolume('gs://neuroglancer/basil_v0/raw_image_cropped', mip=5, progress=True, parallel=True)
print(vol.shape)
vol[:7000,:7000,:]
```

**Comment by william-silversmith** [comment]
```will@wms-cv-benchmark:~$ sudo strace -p 10973
strace: Process 10973 attached
futex(0x258be80, FUTEX_WAIT_BITSET_PRIVATE|FUTEX_CLOCK_REALTIME, 0, NULL, ffffffff


) = ? ERESTARTSYS (To be restarted if SA_RESTART is set)
--- SIGINT {si_signo=SIGINT, si_code=SI_KERNEL} ---
rt_sigreturn({mask=[]})                 = -1 EINTR (Interrupted system call)
write(2, "Traceback (most recent call last"..., 35) = 35
write(2, "  File \"test.py\", line 5, in <mo"..., 38) = 38
open("test.py", O_RDONLY|O_CLOEXEC)     = 25
fstat(25, {st_mode=S_IFREG|0664, st_size=177, ...}) = 0
ioctl(25, TCGETS, 0x7ffd4c0f3a50)       = -1 ENOTTY (Inappropriate ioctl for device)
lseek(25, 0, SEEK_CUR)                  = 0
fcntl(25, F_DUPFD_CLOEXEC, 0)           = 26
fcntl(26, F_GETFL)                      = 0x8000 (flags O_RDONLY|O_LARGEFILE)
fstat(26, {st_mode=S_IFREG|0664, st_size=177, ...}) = 0
read(26, "from cloudvolume import CloudVol"..., 4096) = 177
close(26)                               = 0
lseek(25, 0, SEEK_SET)                  = 0
lseek(25, 0, SEEK_CUR)                  = 0
read(25, "from cloudvolume import CloudVol"..., 8192) = 177
close(25)                               = 0
write(2, "    vol[:7000,:7000,:]\n", 23) = 23
write(2, "  File \"/home/will/cloud-volume/"..., 86) = 86
open("/home/will/cloud-volume/cloudvolume/cloudvolume.py", O_RDONLY|O_CLOEXEC) = 25
fstat(25, {st_mode=S_IFREG|0664, st_size=32378, ...}) = 0
ioctl(25, TCGETS, 0x7ffd4c0f3a50)       = -1 ENOTTY (Inappropriate ioctl for device)
lseek(25, 0, SEEK_CUR)                  = 0
fcntl(25, F_DUPFD_CLOEXEC, 0)           = 26
fcntl(26, F_GETFL)                      = 0x8000 (flags O_RDONLY|O_LARGEFILE)
fstat(26, {st_mode=S_IFREG|0664, st_size=32378, ...}) = 0
read(26, "from __future__ import print_fun"..., 4096) = 4096
close(26)                               = 0
lseek(25, 0, SEEK_SET)                  = 0
lseek(25, 0, SEEK_CUR)                  = 0
read(25, "from __future__ import print_fun"..., 8192) = 8192
read(25, "k_size],\n        \"key\": \"_\".join"..., 8192) = 8192
read(25, "operty\n  def voxel_offset(self):"..., 8192) = 8192
read(25, "ealized_bbox, self.bounds, self."..., 8192) = 7802
close(25)                               = 0
write(2, "    return txrx.cutout(self, req"..., 91) = 91
write(2, "  File \"/home/will/cloud-volume/"..., 74) = 74
open("/home/will/cloud-volume/cloudvolume/txrx.py", O_RDONLY|O_CLOEXEC) = 25
fstat(25, {st_mode=S_IFREG|0664, st_size=15597, ...}) = 0
ioctl(25, TCGETS, 0x7ffd4c0f3a50)       = -1 ENOTTY (Inappropriate ioctl for device)
lseek(25, 0, SEEK_CUR)                  = 0
fcntl(25, F_DUPFD_CLOEXEC, 0)           = 26
fcntl(26, F_GETFL)                      = 0x8000 (flags O_RDONLY|O_LARGEFILE)
fstat(26, {st_mode=S_IFREG|0664, st_size=15597, ...}) = 0
read(26, "from functools import partial\nim"..., 4096) = 4096
close(26)                               = 0
lseek(25, 0, SEEK_SET)                  = 0
lseek(25, 0, SEEK_CUR)                  = 0
read(25, "from functools import partial\nim"..., 8192) = 8192
close(25)                               = 0
write(2, "    handle, renderbuffer = multi"..., 91) = 91
write(2, "  File \"/home/will/cloud-volume/"..., 87) = 87
open("/home/will/cloud-volume/cloudvolume/txrx.py", O_RDONLY|O_CLOEXEC) = 25
fstat(25, {st_mode=S_IFREG|0664, st_size=15597, ...}) = 0
ioctl(25, TCGETS, 0x7ffd4c0f3a50)       = -1 ENOTTY (Inappropriate ioctl for device)
lseek(25, 0, SEEK_CUR)                  = 0
fcntl(25, F_DUPFD_CLOEXEC, 0)           = 26
fcntl(26, F_GETFL)                      = 0x8000 (flags O_RDONLY|O_LARGEFILE)
fstat(26, {st_mode=S_IFREG|0664, st_size=15597, ...}) = 0
read(26, "from functools import partial\nim"..., 4096) = 4096
close(26)                               = 0
lseek(25, 0, SEEK_SET)                  = 0
lseek(25, 0, SEEK_CUR)                  = 0
read(25, "from functools import partial\nim"..., 8192) = 8192
close(25)                               = 0
write(2, "    pool.map(spd, cloudpaths_by_"..., 41) = 41
write(2, "  File \"/usr/lib/python3.5/multi"..., 70) = 70
open("/usr/lib/python3.5/multiprocessing/pool.py", O_RDONLY|O_CLOEXEC) = 25
fstat(25, {st_mode=S_IFREG|0644, st_size=24723, ...}) = 0
ioctl(25, TCGETS, 0x7ffd4c0f3a50)       = -1 ENOTTY (Inappropriate ioctl for device)
lseek(25, 0, SEEK_CUR)                  = 0
fcntl(25, F_DUPFD_CLOEXEC, 0)           = 26
fcntl(26, F_GETFL)                      = 0x8000 (flags O_RDONLY|O_LARGEFILE)
fstat(26, {st_mode=S_IFREG|0644, st_size=24723, ...}) = 0
read(26, "#\n# Module providing the `Pool` "..., 4096) = 4096
close(26)                               = 0
lseek(25, 0, SEEK_SET)                  = 0
lseek(25, 0, SEEK_CUR)                  = 0
read(25, "#\n# Module providing the `Pool` "..., 8192) = 8192
close(25)                               = 0
write(2, "    return self._map_async(func,"..., 69) = 69
write(2, "  File \"/usr/lib/python3.5/multi"..., 70) = 70
open("/usr/lib/python3.5/multiprocessing/pool.py", O_RDONLY|O_CLOEXEC) = 25
fstat(25, {st_mode=S_IFREG|0644, st_size=24723, ...}) = 0
ioctl(25, TCGETS, 0x7ffd4c0f3a50)       = -1 ENOTTY (Inappropriate ioctl for device)
lseek(25, 0, SEEK_CUR)                  = 0
fcntl(25, F_DUPFD_CLOEXEC, 0)           = 26
fcntl(26, F_GETFL)                      = 0x8000 (flags O_RDONLY|O_LARGEFILE)
fstat(26, {st_mode=S_IFREG|0644, st_size=24723, ...}) = 0
read(26, "#\n# Module providing the `Pool` "..., 4096) = 4096
close(26)                               = 0
lseek(25, 0, SEEK_SET)                  = 0
lseek(25, 0, SEEK_CUR)                  = 0
read(25, "#\n# Module providing the `Pool` "..., 8192) = 8192
read(25, "nc, iterable, chunksize=None, ca"..., 8192) = 8192
read(25, "s or pickled'\n              )\n\n "..., 8192) = 8192
close(25)                               = 0
write(2, "    self.wait(timeout)\n", 23) = 23
write(2, "  File \"/usr/lib/python3.5/multi"..., 71) = 71
open("/usr/lib/python3.5/multiprocessing/pool.py", O_RDONLY|O_CLOEXEC) = 25
fstat(25, {st_mode=S_IFREG|0644, st_size=24723, ...}) = 0
ioctl(25, TCGETS, 0x7ffd4c0f3a50)       = -1 ENOTTY (Inappropriate ioctl for device)
lseek(25, 0, SEEK_CUR)                  = 0
fcntl(25, F_DUPFD_CLOEXEC, 0)           = 26
fcntl(26, F_GETFL)                      = 0x8000 (flags O_RDONLY|O_LARGEFILE)
fstat(26, {st_mode=S_IFREG|0644, st_size=24723, ...}) = 0
read(26, "#\n# Module providing the `Pool` "..., 4096) = 4096
close(26)                               = 0
lseek(25, 0, SEEK_SET)                  = 0
lseek(25, 0, SEEK_CUR)                  = 0
read(25, "#\n# Module providing the `Pool` "..., 8192) = 8192
read(25, "nc, iterable, chunksize=None, ca"..., 8192) = 8192
read(25, "s or pickled'\n              )\n\n "..., 8192) = 8192
close(25)                               = 0
write(2, "    self._event.wait(timeout)\n", 30) = 30
write(2, "  File \"/usr/lib/python3.5/threa"..., 60) = 60
open("/usr/lib/python3.5/threading.py", O_RDONLY|O_CLOEXEC) = 25
fstat(25, {st_mode=S_IFREG|0644, st_size=48918, ...}) = 0
ioctl(25, TCGETS, 0x7ffd4c0f3a50)       = -1 ENOTTY (Inappropriate ioctl for device)
lseek(25, 0, SEEK_CUR)                  = 0
fcntl(25, F_DUPFD_CLOEXEC, 0)           = 26
fcntl(26, F_GETFL)                      = 0x8000 (flags O_RDONLY|O_LARGEFILE)
fstat(26, {st_mode=S_IFREG|0644, st_size=48918, ...}) = 0
read(26, "\"\"\"Thread module emulating a sub"..., 4096) = 4096
close(26)                               = 0
lseek(25, 0, SEEK_SET)                  = 0
lseek(25, 0, SEEK_CUR)                  = 0
read(25, "\"\"\"Thread module emulating a sub"..., 8192) = 8192
read(25, "re(self, x):\n        self._lock."..., 8192) = 8192
read(25, "g the internal counter by one.\n\n"..., 8192) = 8192
close(25)                               = 0
write(2, "    signaled = self._cond.wait(t"..., 40) = 40
write(2, "  File \"/usr/lib/python3.5/threa"..., 60) = 60
open("/usr/lib/python3.5/threading.py", O_RDONLY|O_CLOEXEC) = 25
fstat(25, {st_mode=S_IFREG|0644, st_size=48918, ...}) = 0
ioctl(25, TCGETS, 0x7ffd4c0f3a50)       = -1 ENOTTY (Inappropriate ioctl for device)
lseek(25, 0, SEEK_CUR)                  = 0
fcntl(25, F_DUPFD_CLOEXEC, 0)           = 26
fcntl(26, F_GETFL)                      = 0x8000 (flags O_RDONLY|O_LARGEFILE)
fstat(26, {st_mode=S_IFREG|0644, st_size=48918, ...}) = 0
read(26, "\"\"\"Thread module emulating a sub"..., 4096) = 4096
close(26)                               = 0
lseek(25, 0, SEEK_SET)                  = 0
lseek(25, 0, SEEK_CUR)                  = 0
read(25, "\"\"\"Thread module emulating a sub"..., 8192) = 8192
read(25, "re(self, x):\n        self._lock."..., 8192) = 8192
close(25)                               = 0
write(2, "    waiter.acquire()\n", 21)  = 21
write(2, "KeyboardInterrupt\n", 18)     = 18
poll([{fd=5, events=POLLIN}], 1, 0)     = 0 (Timeout)
write(8, "\0\0\0\4\200\3N.", 8)         = 8
futex(0xa50264, FUTEX_WAKE_OP_PRIVATE, 1, 1, 0xa50260, {FUTEX_OP_SET, 0, FUTEX_OP_CMP_GT, 1}) = 1
futex(0xa50220, FUTEX_WAKE_PRIVATE, 1)  = 1
futex(0x7f1398000c10, FUTEX_WAIT_BITSET_PRIVATE|FUTEX_CLOCK_REALTIME, 0, NULL, ffffffff) = 0
futex(0xa50264, FUTEX_WAIT_BITSET_PRIVATE|FUTEX_CLOCK_REALTIME, 33, {1525235592, 443566000}, ffffffff) = -1 EAGAIN (Resource temporarily unavailable)
futex(0xa50220, FUTEX_WAKE_PRIVATE, 1)  = 0
futex(0xa50260, FUTEX_WAKE_PRIVATE, 1)  = 1
futex(0xa50264, FUTEX_WAIT_BITSET_PRIVATE|FUTEX_CLOCK_REALTIME, 35, {1525235592, 443672000}, ffffffff) = -1 EAGAIN (Resource temporarily unavailable)
futex(0xa50220, FUTEX_WAKE_PRIVATE, 1)  = 0
futex(0xa50260, FUTEX_WAKE_PRIVATE, 1)  = 1
futex(0xa50264, FUTEX_WAIT_BITSET_PRIVATE|FUTEX_CLOCK_REALTIME, 37, {1525235592, 443796000}, ffffffff) = -1 EAGAIN (Resource temporarily unavailable)
futex(0xa50220, FUTEX_WAKE_PRIVATE, 1)  = 0
wait4(11375, [{WIFEXITED(s) && WEXITSTATUS(s) == 1}], WNOHANG, NULL) = 11375
wait4(11376, [{WIFEXITED(s) && WEXITSTATUS(s) == 1}], WNOHANG, NULL) = 11376
wait4(11377, [{WIFEXITED(s) && WEXITSTATUS(s) == 1}], WNOHANG, NULL) = 11377
wait4(11378, [{WIFEXITED(s) && WEXITSTATUS(s) == 1}], WNOHANG, NULL) = 11378
wait4(11379, [{WIFEXITED(s) && WEXITSTATUS(s) == 1}], WNOHANG, NULL) = 11379
wait4(11380, [{WIFEXITED(s) && WEXITSTATUS(s) == 1}], WNOHANG, NULL) = 11380
wait4(11381, [{WIFEXITED(s) && WEXITSTATUS(s) == 1}], WNOHANG, NULL) = 11381
wait4(11382, [{WIFEXITED(s) && WEXITSTATUS(s) == 1}], WNOHANG, NULL) = 11382
wait4(11383, [{WIFEXITED(s) && WEXITSTATUS(s) == 1}], WNOHANG, NULL) = 11383
wait4(11384, [{WIFEXITED(s) && WEXITSTATUS(s) == 1}], WNOHANG, NULL) = 11384
wait4(11385, [{WIFEXITED(s) && WEXITSTATUS(s) == 1}], WNOHANG, NULL) = 11385
wait4(11386, [{WIFEXITED(s) && WEXITSTATUS(s) == 1}], WNOHANG, NULL) = 11386
wait4(11387, [{WIFEXITED(s) && WEXITSTATUS(s) == 1}], WNOHANG, NULL) = 11387
wait4(11388, [{WIFEXITED(s) && WEXITSTATUS(s) == 1}], WNOHANG, NULL) = 11388
wait4(11389, [{WIFEXITED(s) && WEXITSTATUS(s) == 1}], WNOHANG, NULL) = 11389
wait4(11390, [{WIFEXITED(s) && WEXITSTATUS(s) == 1}], WNOHANG, NULL) = 11390
rt_sigaction(SIGINT, {SIG_DFL, [], SA_RESTORER, 0x7f13c9926390}, {0x63f1d0, [], SA_RESTORER, 0x7f13c9926390}, 8) = 0
munmap(0x7f13c9d52000, 32)              = 0
munmap(0x7f13c9d53000, 32)              = 0
close(5)                                = 0
close(8)                                = 0
munmap(0x7f13c9b6b000, 32)              = 0
munmap(0x7f13c9d51000, 32)              = 0
close(7)                                = 0
close(6)                                = 0
munmap(0x7f13ad09e000, 262144)          = 0
munmap(0x7f13ad0de000, 262144)          = 0
munmap(0x7f13af358000, 262144)          = 0
munmap(0x7f13af398000, 262144)          = 0
munmap(0x7f13af3d8000, 262144)          = 0
munmap(0x7f13af418000, 262144)          = 0
munmap(0x7f13af458000, 262144)          = 0
munmap(0x7f13af498000, 262144)          = 0
munmap(0x7f13af4d8000, 262144)          = 0
munmap(0x7f13b1b59000, 262144)          = 0
munmap(0x7f13b1b99000, 262144)          = 0
munmap(0x7f13b1bd9000, 262144)          = 0
munmap(0x7f13b1c19000, 262144)          = 0
munmap(0x7f13b1c59000, 262144)          = 0
munmap(0x7f13b1c99000, 262144)          = 0
munmap(0x7f13c9d55000, 32)              = 0
munmap(0x7f13c3a39000, 262144)          = 0
munmap(0x7f13c9d54000, 32)              = 0
getsockname(3, {sa_family=AF_INET, sin_port=htons(51252), sin_addr=inet_addr("10.240.0.31")}, [16]) = 0
getpeername(3, {sa_family=AF_INET, sin_port=htons(443), sin_addr=inet_addr("74.125.141.84")}, [16]) = 0
close(3)                                = 0
getsockname(4, {sa_family=AF_INET, sin_port=htons(54038), sin_addr=inet_addr("10.240.0.31")}, [16]) = 0
getpeername(4, {sa_family=AF_INET, sin_port=htons(443), sin_addr=inet_addr("74.125.141.95")}, [16]) = 0
close(4)                                = 0
munmap(0x7f13c3a79000, 262144)          = 0
munmap(0x7f13b1cd9000, 262144)          = 0
munmap(0x7f13c2886000, 262144)          = 0
munmap(0x7f13c2806000, 262144)          = 0
munmap(0x7f13c2946000, 262144)          = 0
munmap(0x7f13c39b9000, 262144)          = 0
munmap(0x7f13bbb4b000, 262144)          = 0
munmap(0x7f13c2906000, 262144)          = 0
munmap(0x7f13b64e8000, 262144)          = 0
munmap(0x7f13c2dd0000, 262144)          = 0
munmap(0x7f13b6428000, 262144)          = 0
munmap(0x7f13c033f000, 33554432)        = 0
munmap(0x7f13be33f000, 33554432)        = 0
munmap(0x7f13bc33f000, 33554432)        = 0
munmap(0x7f13b9b3e000, 33554432)        = 0
munmap(0x7f13b6b3c000, 33554432)        = 0
munmap(0x7f13b433b000, 33554432)        = 0
munmap(0x7f13b233b000, 33554432)        = 0
munmap(0x7f13afb3a000, 33554432)        = 0
munmap(0x7f13ad339000, 33554432)        = 0
munmap(0x7f13aab38000, 33554432)        = 0
munmap(0x7f13a7b36000, 33554432)        = 0
munmap(0x7f13a4b34000, 33554432)        = 0
munmap(0x7f13a2b34000, 33554432)        = 0
munmap(0x7f13a079b000, 33554432)        = 0
munmap(0x7f139e75b000, 33554432)        = 0
exit_group(1)                           = ?
+++ exited with 1 +++```

**Comment by william-silversmith** [comment]
Looks like this is intended behavior in mp.Pool if a child process dies suddenly. 

https://nelsonslog.wordpress.com/2015/01/20/multiprocessing-worker-processes-dying/

We addressed this issue in #108
